### PR TITLE
Fix home page package issues

### PR DESCRIPTION
**Pull request by Fox32**
The dependencies for `@types/react` and `react-router` differ from all other packages. Also, following the readme to add it doesn't work as there is a mistake in the readme.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

File: plugins/home/README.md [hunk 1]
```diff
@@ -43,9 +43,14 @@ import { HomepageCompositionRoot } from '@backstage/plugin-home';
 import { HomePage } from './components/home/HomePage';
 
 // ...
-<Route path="/" element={<HomepageCompositionRoot />}>
-  <HomePage />
-</Route>;
+<Route
```
Comment (emmaindal) [inline plugins/home/README.md:46]: Hey @Fox32, versions looks good to me. But looking at the changes of the docs I'm curious in what does not work here? Looking at the [App](https://github.com/backstage/backstage/blob/master/packages/app/src/App.tsx#L117) the example in the readme is equal to how the element prop and children is used there.

Comment (emmaindal) [reply plugins/home/README.md:46]: I was also trying setting up it myself now using the docs and adding the following to the App.

```
<Route path="/" element={<HomepageCompositionRoot />}>
  <HomePage />
</Route>
```

which worked well. But happy to make these changes if required, I just want to make sure there is nothing else which is the root cause to why it doesn't work as expected!

Comment (Fox32) [reply plugins/home/README.md:46]: You are right, it should actually work and it works for other plugins. But once I do it in our app, I miss all the widgets from the homepage:
image

Would be fine to keep it as is for now and just fix it locally on our side. But I would still remove the semicolon 😉

Comment (Fox32) [reply plugins/home/README.md]: Arg, pettier always includes the semicolon, anyway let's keep it as-is.

**Review by emmaindal — APPROVED** [review]

File: plugins/home/README.md (hunk 1 above)
Comment (emmaindal) [reply plugins/home/README.md]: interesting, do you now why it doesn't give you the widgets?

Comment (Fox32) [reply plugins/home/README.md]: No idea, we use the same pattern for the scaffolder, explore and search, no problems there.

Comment (emmaindal) [reply plugins/home/README.md]: Alright, the only thing I can think of is looking into your HomePage component in the app which should live under `packages/app/src/components/home/HomePage.tsx` if I look at [the docs](https://github.com/backstage/backstage/blob/master/plugins/home/README.md#setting-up-the-home-page), what that looks like and if its missing the widgets!

**Pull request merged**